File: java/01_TwoSum.java
public int[] twoSum(int[] nums, int target) {
    HashMap<Integer, Integer> arr = new HashMap<>();
    int[] result = new int[2];
    for (int i=0; i < nums.length; i++) {
        int numberFind = target - nums[i];
        if (arr.containsKey(numberFind)) {
            result[0] = arr.get(numberFind);
            result[1] = i;
            break;
        } else {
            arr.put(nums[i], i);
        }
    }
    return result;
}

File: swift/01_TwoSum.swift
func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var result:[Int] = []
    var dict = [Int: Int]()
    
    for (index, num) in nums.enumerated() {
        let complement = target - num
        if (dict.keys.contains(complement)) {
            result.append(index)
            result.append(dict[complement]!)
        }
        dict[num] = index
    }
    return result
}
